README.md
eob1-waterdeep
==============

Eye of the Beholder: Waterdeep Sewers mod

CHANGELOG.txt
v0.02.19 2015-02-14
changes made by Carlos
levels 1-12 are playable
all the puzzles, keys, doors, special quests etc. should work as original
or as close to original as possible
dwarvish writings now readable only by Taghor, Dohrum and Keirgar 


v0.02.18 
changes made by Carlos
Levels 7 - 9 playable
few minor things are missing 


v0.02.17 (changes made by Carlos)
==========
 LEVEL 1
==========
no changes


==========
 LEVEL 2
==========
- updated script_entity_2 (at possiton 1,1)  part function SpecialQuest2(sender)
should spawn rations at every location of dagger - not just one at party location
- all drainpipes added with text "This slimy, smelly drainpipe reveals nothing."
- removed eob_wall_button_7  on 10,21
- removed eob_wall_button_8  on  2,21
- removed eob_wall_button_9  on  7,18
replaced by pull chains 
in original there is small blue button from each side of the door
pull chain is not button but still makes more sense then just one button outside

- removed monster eob_flind1_1_1 on 23,6
there are just two monsters in original in this section


==========
 LEVEL 3 
==========
-museum/galerie with monsters fixed - they attack only when attacked first
see monsters.lua cloneObject{  name = "eob_kuotoa1_1_galery"
-replaced monsters as per original
-added hidden preasure plate 11,18 as per original
-added silent teleporter & rotator 15,19 as per original
- fixed door button on 17,19
- fixed door lever  on 16,22
- added hidden plate on 17,12 to close iron door eob_sewers_door_metal_17 on 17,12
- lock at 25,13, added spawner of two monsters 
to match with original (two Kuo_toa)
- red gem alcoves missing on empty click
"THIS HOLE LOOKS LIKE THE INSIDE OF A JEWEL BOX."
possitons 19.17 and 20,23
attempt to replace with hidden plates, shown only until you place gem inside
- pit room entrance 21,23
- added new red gem (as per original) to 17,24 and torch 18,24
- attempt to fix the pit system in south east room
- in original they got two modes, which would require a LOT of scripting
- added hidden plate on 21,24 to reset the pits when entering room
- fixed exit teleporter on 29,25 under pit room as per original
- added two new teleports in the end area on 4,5 and 11,12 as per original
- special secret level 3 fixed 
- added the object secret 10,6 now it plays sound when eyes are removed
- added red gem in 14,28 (spawns when red gem is used on lock on 19,17) as per original (eventhough you end up with useless red gem this way)


==========
 LEVEL 4 
==========
- when going from level 4 to level 3 one ends up at wrong locations
  removed stairs up from lvl4 that leads to lvl3, replaced with ladder & ceiling shaft & teleporter
  I know in original there are stairs, but this looks good and what is more important WORKS 
- fixed Tahorn, he is NOT an insectoid mage guys, tried to balance his numbers based on original + what other characters in grimrock are at lvl3
- button at 16,21 now also opens secret wall at same location to room with spider (cant find secret wall that would match color with other walls)
- fixed spider (now he guard/waits behind the secrete dwarf cube at 18,21)
- fixed text on wall 20,20
- fixed text for levers to 'Access control lever.'
- fixed 3 levers to 3 holes, fixed preasure plate to toggle the holes as per original
- fixed texts on wall in 11,11 room
- fixed text possition on 10,20
- fixed plate activation toggles the hole on 8,20 as per original
- fixed stairs between 4th and 5th level, those on 5th moved one square down, now looks much better
- fixed button at 14,26 to toggle door at 13,26 (not just open once)
- moved door from 13,1 to 14,1 (cosmetic change) and attach lever on 15,2 to them 
- add if both 9,13 and 13,13 doors are closed, remove secrete cube wall on 11,18 as per original
- add secret cube wall on 8,11 toggled by chain on 11,18
- fixed mechnism of 4 doors at 3,7, any door opening when all doors are closed spawns one potion, up to 3 potions (defined by counter)
- moved pressure plates closer to door in spider sector to close them right when you enter on 13,27
- add one spide at 25,25
- removed dwarf cube from 26,18 as per original
- add dwarf cube to 27,17 as per original
- dwarf cubes at 26,13 and 27,17, set with counters, each needs to be pressed 2 times to open its cube, at least 10 sec apart (timer)
  unable to fetch the exact sequence from original
  but it seems that the plates that activates them must be pressed several times 
  its is definitely possible to open both by going south and north 2 times
  I also saw in original that only 27,17 was open - unable to reproduce yet though
- connected lever on 3,3 secret area to cube 16,4 (toggle) and removed the eob_ruins_wall_small_statue 16,3/16,5 (as the graphic doesnt fit to cube wall)
- added pressure_plate_release to wall moves at 16,10
- replaced eob_ruins_illusion_wall on 5,4 with eob_ruins_illusion_wall_rune_fake (which dont disapper once walked through)
- added system of spider respawning and each spider passing a web will renew that web and eggs now
- fixed oracle of knowledge (item identifier if you put orb of power into alcove )
- add timer to replace hacked webs & eggs with fixed ones + spider spawner

done differently then original
- added secret when discovered illusional wall at 5,4
- drow clever at 11,18 is now spawned by special quest (otherwise the special quest on this level does nothing)


==========
 LEVEL 5 
==========
- fixed secret button on 2,15 and secret wall on 3,15 but doesnt fit together yet
- removed pull chain from door 1,13 (in original they are not openable from south, only from north), temporary added button on nearby wall
- add pressure plate on 10,5 with secret
- add dwarven cube on 19,16 and connected to button on 20,16
- door on 8,10 should be forceable
- removed pull chain on door 6,10, attached to secret button on 6,9
- fixed alcove 10,13 - special quest for level 5, on inserting 5 eob_rations, turn them into 5 eob_rations_iron

mod_assets/scripts/notes.lua
local WentlysNote = {}

-- level 1
WentlysNote[1] = { text = [[
My name is Wently Kelso, a famous archeologist. 
During this expedition to the deepest depths of
the sewers of Waterdeep, I'll write notes for my
latest book entitled "The World Beneath Waterdeep".

Our guide on expedition is a mangy humanoid of
indeterminate species. I can't pronounce his real
name, so I call him Bennet, in honor of my mangy
half-brother on my mothers side of the family. 
As we followed Bennet into the tunnels and catacombs
of the sewer system, we can hear the soft foot
steps of large dog-like creatures known as Kobolds.
]], l = 1, x = 11, y = 15, f = 0 }

WentlysNote[2] = { text = [[
This area was used to control the water flow
through the first level of the sewer system.
If the water flowed from the north, certain
passageways would close to prevent the water
from flooding the level. The same would happen
if the water flowed from the east.
]], l = 1, x = 15, y = 12, f = 1 }

WentlysNote[3] = { text = [[
According to the city sewer layout, some of the
doors require a constant flow of water over their
pressure plate to remain open. Once the water
stopped flowing over the pressure plate, the door
would close, to prevent backwash.
]], l = 1, x = 21, y = 18, f = 0 }

WentlysNote[4] = { text = [[
Our guide, Bennet, pointed out a secret door here.
He tried (with our help) to bash the door in with
his head but was unsuccessful. There must be a 
secret button or lever around here somewhere.
]], l = 1, x = 25, y = 22, f = 1 }

WentlysNote[5] = { text = [[
There is a depression in the north wall with rolled
up papers inside. I am surprised that the papers
survived. I would have thought that unprotected
paper would rot in the sewer.
]], l = 1, x = 12, y = 25, f = 1 }

WentlysNote[6] = { text = [[
Here we found an emergency flood control door.
It is designed so that if the area flooded any
trapped workers could open the door by pushing
the button. The water would rush down into the
second level saving the workers.
]], l = 1, x = 16, y = 24, f = 2 }

-- level 2
WentlysNote[7] = { text = [[
We looked down the long dark hole that led to the
second level of the sewer system. The smell of
rotting flesh filled our nostrils and we decided
that the Bennet, our guide, should go down first.
Looking down into the dark, he became frightened
and stopped about five feet down. We encouraged him
to continue on by dropping rocks down the hole.
]], l = 2, x = 19, y = 23, f = 1 }

WentlysNote[8] = { text = [[
This door was stuck partially open. Bennet gripped
the bottom of the door and pulled up as hard as he
could. By his whining I assume that he has somehow
hurt himself.
]], l = 2, x = 19, y = 18, f = 0 }

WentlysNote[9] = { text = [[
Our guide was overcome by sewer gas around here
somewhere. He spun around a couple of times and
ended up facing the wrong direction.
]], l = 2, x = 19, y = 26, f = 1 }

WentlysNote[10] = { text = [[
This area is an emergency exit. If the water level
became too high for the workers to get out, they
would step into this area and it would teleport
them to the surface. Unfortunately, the teleporter
has broken down and it will only teleport you to
another area within the sewer system. I don't know
how we will we ever find our passage out of here.
]], l = 2, x = 24, y = 28, f = 3 }

WentlysNote[11] = { text = [[
When our guide read "R.A.T.S." he ran in terror.
We had to track him down and drag him back.
The city sewer map claims that R.A.T.S. stands for
Rapid Access Teleport System. 
I wonder where it will teleport you to.
]], l = 2, x = 9, y = 25, f = 1 }

WentlysNote[12] = { text = [[
This area appears to be a shuttle of some kind.
We pushed the button and the door closed. Then,
we pushed the button again and there was a strange
sound. 
What does the inscription on the side wall mean?
]], l = 2, x = 10, y = 21, f = 2 }

WentlysNote[13] = { text = [[
We were warned by our guide not to push the button
on the wall in the cell of north-west room.
What does our guide know? I pushed it anyway.
I wonder what will happen.
]], l = 2, x = 13, y = 12, f = 3 }

WentlysNote[14] = { text = [[
This keyhole in north-west corridor is different
than the others we have run into thus far.
Does it require a special key?
]], l = 2, x = 22, y = 8, f = 1 }

-- level 3
WentlysNote[15] = { text = [[
The stench of sewer gas makes our heads spin.
Our guide, Bennet, has informed us that this is
the perfect environment for the kuo-toa.
We will have to keep our guard up.
]], l = 3, x = 26, y = 5, f = 1 }

WentlysNote[16] = { text = [[
There is a forest of pits and pres-
sure plates in this room. Perhaps if we
trigger the pressure plates one at a time,
we can explore the whole room.
However, I do wonder what is down the
pits. Perhaps we could throw Bennet
down there and have him take a look.
]], l = 3, x = 21, y = 14, f = 2 }

WentlysNote[17] = { text = [[
Bennet translated the rune on the
wall here. It reads, "Museum". 
Imagine, a museum down here. How nice. 
I just love museums.
]], l = 3, x = 11, y = 17, f = 3 }

WentlysNote[18] = { text = [[
Just as we thought our adventure was
coming to an end, we found a stairway
leading down to another level. 
The stone work on this level looks nothing 
like the sewers above. Could these stairs 
lead to the fabled dwarven ruins? 
Our hopes are high.
]], l = 3, x = 8, y = 9, f = 1 }

-- level 4

WentlysNote[19] = { text = [[
There is an interesting dwarven statue
on the far wall. It is in the shape 
of a gargoyle with one arm raised.
Our guide pulled the gargoyle's arm 
and nothing seemed to happen. 
Those dwarves are such jokers.
]], l = 4, x = 21, y = 8, f = 1 }


WentlysNote[20] = { text = [[
We continue going south. 
Some of the landmarks are starting 
to look familiar.
I take this as a good sign. 
However, I don't know how many 
more spider bites our guide can take.
]], l = 4, x = 30, y = 5, f = 1 }


-- level 5
WentlysNote[21] = { text = [[
Our guide has informed us that the
rune on the wall reads, "Safe passage".
We decided to see how safe it was by
pushing our guide into the wall.
Amazingly, he passed right through.
]], l = 5, x = 9, y = 24, f = 0 }

WentlysNote[22] = { text = [[
Our guide has informed us that the
rune on the wall reads, "Safe passage".
We decided to see how safe it was by
pushing our guide into the wall.
Amazingly, he passed right through.
]], l = 5, x = 9, y = 24, f = 0 }

WentlysNote[23] = { text = [[
There are dwarves down here! One
dwarf named Armun, claimed that this
area was his lost kingdom. He asked us to
help them find Keirgar, a prince, that had
been captured by the dwarves' enemies.
I told them that we would do what we
could to help them with their little quest.
]], l = 5, x = 14, y = 10, f = 1 }

WentlysNote[24] = { text = [[
We found a very nice dwarf here that
tended to a cut I received on my finger.
Our guide, Bennet, asked the dwarf to
look at a large gash on his head. It is just
so hard to find good help these days.
]], l = 5, x = 18, y = 7, f = 1 }

WentlysNote[25] = { text = [[
We were told by the dwarves to
be careful of this area. It was used 
centuries ago to teleporl materials 
to the surface. 
Now, the teleporters seem to be
interconnected.
]], l = 5, x = 29, y = 9, f = 0 }

-- level 6

WentlysNote[26] = { text = [[
This area was used for diplomatic
meetings. The Dwarves would stand on
one side and of the pressure plate and their
enemy would stand on the other side.
Both parties would lay down their
weapons, and the talks would begin.
]], l = 6, x = 25, y = 4, f = 3 }

WentlysNote[27] = { text = [[
We met a dark-robed person. He was
very unfriendly and would not give us so
much as his name. He just stood there
glaring at us. I would have challenged
him, if we were not in such a hurry.
]], l = 6, x = 26, y = 14, f = 1 }

WentlysNote[28] = { text = [[
I think that we found an example of
the classic dwarven "dart trap". In such a
trap, you step on a pressure plate and
darts fly out of the holes. To test my
theory I had our brave guide, Bennet,
stand in front of the holes, while I step
on the plate.
]], l = 6, x = 5, y = 12, f = 0 }

-- level 7
WentlysNote[29] = { text = [[
When we stepped on this pressure
plate we heard a whooshing sound.
Luckily, we all ducked behind our guide,
Bennet, just before the explosion.

As we moved forward we heard the
whooshing sound again. Again, everyone
ducked but Bennet. He doesn't look much
the worse for wear, however.
]], l = 7, x = 6, y = 4, f = 0 }


WentlysNote[30] = { text = [[
I removed a dart from our guide's
chest and placed it on the shelf, but 
nothing happened. Our guide, Bennet,
skulked away and is now sniffing at
something over on the north wall.
]], l = 7, x = 20, y = 14, f = 3 }

-- level 8 
WentlysNote[31] = { text = [[
I think that Bennet has been under-
ground for too long, he is urging us to go
back the way we came.

Bennet is causing trouble again, he
still wants us to turn around and go back
where we came.
]], l = 8, x = 3, y = 30, f = 0 }

WentlysNote[32] = { text = [[
Bennet is looking around as if
he is confused. This does not surprise me,
for I believe that his brain is only slightly
larger than that of a grub fish. 
We go on despite Bennet's confusion.
]], l = 8, x = 1, y = 12, f = 0 }


WentlysNote[33] = { text = [[
Our guide read the message on the
wall, and became confused again.
I believe that the value of an education
is wasted on Bennet.
]], l = 8, x = 14, y = 6, f = 0 }


WentlysNote[34] = { text = [[
I told our guide, Bennet, not to touch
the spider button on the west wall, but he
accidently slipped and hit it with his
head. There was a strange sound and sud-
denly, we were attacked by many hell
hounds. The party ran up the hall to see if
the path was clear, while Bennet stayed
behind to keep the hell hounds in check.
The last utterance I heard from our guide
was, "Sit boy! Sit!"
]], l = 8, x = 5, y = 1, f = 3 }

-- level 9
WentlysNote[35] = { text = [[
We have lost our faithful guide, Bennet.
No doubt, he is back on the surface, 
probably, at the Yawning Portal Inn, 
enjoying a nice mutton pie and a pint of ale.
We, on the other hand are forced to continue
onward and downward.
]], l = 9, x = 22, y =4 , f = 1 }

WentlysNote[36] = { text = [[
I found a dagger in a niche on the
wall. I am pleased it is an exact copy of
the dagger I was carrying. But, when I put
the new dagger in my belt pouch, I seem
to have misplaced my old dagger. Perhaps
it was with Bennet when he disappeared.
]], l = 9, x = 25, y = 9, f = 0 }

WentlysNote[37] = { text = [[
We stepped into this area and darts
started flying at us. We no longer 
had a guide to absorb the attack, 
so we ran as fast as we could down the hall.
]], l = 9, x = 7, y = 11, f = 1 }


--level 10
WentlysNote[38] = { text = [[
To the west is a forest of teleporters
and levers. Another problem with 
a binary solution, we either make it
through the teleporters, or we do not.
]], l = 10, x = 19, y = 30, f = 0 }

WentlysNote[39] = { text = [[
We found our guide, Bennet, chained
to the wall with a dwarf. Too bad,
we only had time to rescue our guide, 
but I am sure some other adventurers 
will come along and rescue the dwarf.
]], l = 10, x = 17, y = 28, f = 0 }

WentlysNote[40] = { text = [[
We have searched the whole level
and this seems to be the only way down
to the lower levels. We looked down 
the pit and could not see the bottom,
so I pushed our guide into it. 
There was an almost immediate thump. 
It cannot be that deep.
]], l = 10, x = 14, y = 11, f = 0 }


-- level 11
WentlysNote[41] = { text = [[
When will this ever end? When will we
find our way out? I am afraid that we will
never return to the surface and that no
one will ever read this marvelous work.
]], l = 11, x = 15, y = 11, f = 3 }

WentlysNote[42] = { text = [[
Here we found a hallway of levers.
"Never leave a lever down!",
my mother always told me.
]], l = 11, x = 24, y = 9, f = 1 }

WentlysNote[43] = { text = [[
Our guide, Bennet, is continuously 
confused as we move around these
halls. We seem to be circling endlessly,
but we also seem to be finding new
things as we pass. 
I am not sure how we missed these things 
the first times around.
]], l = 11, x = 1, y = 5, f = 0 }


--level 12
WentlysNote[44] = { text = [[
We have met the evil crime lord
Xanathar and he will not let us leave.
We are doomed to stay here forever.

The walls in this area are beautiful.
The ornamentation is exquisite. 
If we are doomed to stay here forever, 
at least we can do so in civilized surroundings.
]], l = 12, x = 22, y = 7, f = 0 }

WentlysNote[45] = { text = [[
When we stepped into this room,
three stone golems came out to greet us.
Our guide somehow offended them 
and they attacked us. 
Of course, we ran away.
]], l = 12, x = 10, y = 21, f = 0 }

WentlysNote[46] = { text = [[
Our guide, Bennet, crossed a light
beam in this area and was hit in the head
by a giant fireball. For a brief moment, 
he looked just like a dwarven torch.
]], l = 12, x = 29, y = 10, f = 0 }





cloneObject{
	name = "wentlys_note",
	baseObject = "note",
	uiName = "Wently's note",
}

cloneObject{
	name = "WentlysNoteSpawner",
	baseObject = "dungeon_door_metal",
	onOpen = function()
		for id,wnote in pairs(WentlysNote) do
			spawn("wentlys_note", wnote.l, wnote.x, wnote.y, wnote.f ):setScrollText("Note #"..id.."\n--------------\n"..wnote.text.."\n~Wently Kelso")
		end
		fw.debugPrint("Wentlys Notes spawned ("..#WentlysNote..")")
		spawn("counter", 1, 1, 2, 0, "wentlys_note_total"):setValue(#WentlysNote)
	end
}
